storage/innobase/row/row0merge.cc

```diff
@@ -2286,6 +2303,48 @@ row_merge_read_clustered_index(
 			}
 		}
 
+		if (old_table->versioned()) {
+			if (new_table->versioned() && !drop_historical) {
```

Comment:
Why the condition on `drop_historical`? Shouldn’t `new_table->versioned()` imply `!drop_historical`?
Or shouldn’t `drop_historical` be equal to `old_table->versioned() && !new_table->versioned()`? If not, why not?

Reply:
`drop_historical` is for `versioning_alter_history=drop`.